### PR TITLE
New: Add rule no-angle-bracket-type-assertion

### PR DESCRIPTION
**Pull request by weirdpattern**
This includes
- New rule no-angle-bracket-type-assertion
- Corresponding test
- Corresponding documentation
- Fix valid scenario in type-annotation-spacing (was failing)
- Update to README.md to include the new rule

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "eslint": "~3.7.0",
     "eslint-config-eslint": "^3.0.0",
     "mocha": "^2.4.5",
-    "typescript-eslint-parser": "^0.4.0"
+    "typescript-eslint-parser": "^1.0.0"
```
Comment (dannyfritz) [inline package.json]: This should probably go in a separate PR.

Comment (weirdpattern) [inline package.json]: @dannyfritz You are right, I was working on some other rule that required the upgrade.
I will send a different PR for this.

File: docs/rules/no-angle-bracket-type-assertion.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Enforces the use of `as Type` assertions instead of `<Type>` assertions.
+
+Typescript disallows the use of `<Type>` assertions in `.tsx` because of the similitude with  
+JSX's syntax, which makes it impossible to parse. 
+
+## Rule Details
+
+This rule aims to standardise the the use of type assertion style across the codebase
```
Comment (JamesHenry) [inline docs/rules/no-angle-bracket-type-assertion.md]: Typo: Double "the"

File: tests/lib/rules/type-annotation-spacing.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ ruleTester.run("type-annotation-spacing", rule, {
             parser: "typescript-eslint-parser"
         },
         {
-            code: "function foo(a:string) {}",
+            code: "function foo(a: string) {}",
```
Comment (JamesHenry) [inline tests/lib/rules/type-annotation-spacing.js]: Please remove this from this PR, it should be addressed separately

File: docs/rules/no-angle-bracket-type-assertion.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Enforces the use of `as Type` assertions instead of `<Type>` assertions.
+
+Typescript disallows the use of `<Type>` assertions in `.tsx` because of the similitude with
```
Comment (JamesHenry) [inline docs/rules/no-angle-bracket-type-assertion.md]: Please use "PascalCase" for "TypeScript".

"Similitude" is an epic, but somewhat archaic word, I feel it would be much more common to use "similarity" 😄

**Comment by weirdpattern** [comment]
Done, I will submit a new PR for the type-annotation-spacing.
